.github/workflows/stable.yml

```diff
@@ -6,33 +6,44 @@ jobs:
   test:
     runs-on: ubuntu-latest
 
+    env:
+      BUNDLE_WITHOUT: "benchmark"
+      JRUBY_OPTS: "--debug"
+      # $BUNDLE_GEMFILE must be set at the job level, so it is set for all steps
+      BUNDLE_GEMFILE: ${{ github.workspace }}/gemfiles/${{ matrix.gemfile }}.gemfile
+
     strategy:
       matrix:
-        ruby: [2.5.8, 2.6.6, 2.7.2, 3.0.0, jruby-9.2.14.0]
-
+        gemfile:
+          - rails70
+        ruby:
+          - 2.7
+          - "3.0"
+          # 3.1 has test failures, but no installation failures.
+          # - 3.1
+        include:
+          - ruby: 2.6
+            gemfile: rails60
+          - ruby: 2.7
+            gemfile: rails60
+          - ruby: 2.6
+            gemfile: rails61
+          - ruby: 2.7
+            gemfile: rails61
+          - ruby: 2.6
+            gemfile: rails52
```

Comment:
the gemfiles surprise me a bit... do we need them? The only full on rails usage should be in cucumber tests as separate tests/files/projects. Am I missing something? I would like to avoid our test suite having rails loaded at all times, with all their monkey patches etc. it could make it easy to break for us when working with normal ruby projects :thinking:

Reply:
In order to be able to pick an active* version that avoids 7.0's high Ruby version requirement of 2.7 (no released JRuby supports 2.7, now, does it?). 

Happy to tighten/reduce impact area in some way. Perhaps a conditional in the Gemfile, using different version requirements for different Ruby versions?

This is just me sharing some guesses!

Reply:
I don't only mean these but also rails 7.0 - like iirc the only thing that uses rails is a specific rspec_rails test project in the test projects folder that is currently defined as rails 6.1 iirc - nothing else should require any kind of rails :thinking: 

And I'm fine with that only testing 6.1, if I wanna test 7.0 I can write another test and another sample project - no need to run matrixes against entire rails versions SImpleCov is thankfully far enough away from Rails to not depend on it like that :D